### PR TITLE
feat(flare): add diagnose datadog-connectivity output to the flare archive

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

This PR adds the output of `diagnose datadog-connectivity` command in the flare archive with some code refactor in the `archive.go` file. 

### Motivation

Having more information in the flare for troubleshooting 

### Describe how to test/QA your changes

- Run `datadog-agent --no-color diagnose datadog-connectivity` and `datadog-agent diagnose datadog-connectivity` and verify that you get an output.
- Run `datadog-agent flare` and verify that there is a new `connectivity.log` file, in the archive, containing the same output as the command above, and also that the other files are still here with a content similar to before.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/flare/archive.go [hunk 1]
```diff
@@ -694,6 +700,22 @@ func zipDiagnose(tempDir, hostname string) error {
 	return writeScrubbedFile(f, b.Bytes())
 }
 
+func zipDatadogConnectivity(tempDir, hostname string) error {
+	var b bytes.Buffer
+
+	writer := bufio.NewWriter(&b)
+	connectivity.RunDatadogConnectivityDiagnose(writer, false) //nolint:errcheck
```
Comment (ogaca-dd) [inline pkg/flare/archive.go:707]: Why ignoring the error?

File: pkg/flare/archive.go [hunk 2]
```diff
@@ -694,6 +700,22 @@ func zipDiagnose(tempDir, hostname string) error {
 	return writeScrubbedFile(f, b.Bytes())
 }
 
+func zipDatadogConnectivity(tempDir, hostname string) error {
```
Comment (ogaca-dd) [inline pkg/flare/archive.go:703]: Bonus: The code of this function is very similar to several other functions like `zipClusterAgentDiagnose`, `zipSecrets` or `zipDiagnose`. You can factorize them.

File: pkg/diagnose/connectivity/hooks.go [hunk 3]
```diff
@@ -51,29 +53,36 @@ var (
 	tlsColorFunc = color.YellowString
 )
 
+// writeWrapper is used to pass a io.Writer variable to the hooks so that they
+// can output into a terminal if the command is used via CLI or into a buffer
+// when requesting a flare
+type writerWrapper struct {
```
Comment (ogaca-dd) [inline pkg/diagnose/connectivity/hooks.go:59]: I think the name can be improved to better reflect what struct does, ie implements hooks for `ClientTrace`. You can name it `connectivityHooks` for example.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/diagnose/connectivity/hooks.go (hunk 3 above)
Comment (Kaderinho) [reply pkg/diagnose/connectivity/hooks.go]: yeah you're right. Done here 14970e4d79ac03d8b2826fc6bf5a3579c9c0c157

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/flare/archive.go [hunk 4]
```diff
@@ -749,23 +731,15 @@ func zipVersionHistory(tempDir, hostname string) error {
...
 func zipConfigCheck(tempDir, hostname string) error {
-	var b bytes.Buffer
-
-	writer := bufio.NewWriter(&b)
-	GetConfigCheck(writer, true) //nolint:errcheck
-	writer.Flush()
-
-	return writeConfigCheck(tempDir, hostname, b.Bytes())
+	fct := func(w io.Writer) error {
+		return GetConfigCheck(w, true)
```
Comment (ogaca-dd) [inline pkg/flare/archive.go:735]: Note: the error was previously ignored.

**Pull request merged**